### PR TITLE
auto slide function +  parameterisation

### PR DESCRIPTION
**Pull request by christophsaile**
Hey,
i added the following features:

- comment/docu for all the possible parameters
- autoplay function with different options (duration, stop on mouseover ...)
- parameterisation of easing effect and slide speed

#39

### CODE REVIEW

File: src/components/WcSlider/scaffolding/WcSlider.hbs [hunk 1]
```diff
@@ -1,29 +1,53 @@
...
+		'[{"src" : "string", "bp" : number},
+		{"src" : "string", "bp" : number},
+		{ .... },
+		{"src" : "string"}]'
+--}}
+
+<wc-slider has-preview has-counter 
+	has-autoplay='[{"autoplay" : true },
+		{"stopOnMouseOver" : true },
+		{"autoplaySpeed" : 7000 }]'
```
Comment (no-creative-name) [inline src/components/WcSlider/scaffolding/WcSlider.hbs]: When an attribute is called "has-autoplay", I would expect it to be a boolean and not an object. I would suggest "autoplay-options".

File: src/components/WcSlider/scaffolding/WcSlider.hbs [hunk 2]
```diff
@@ -1,29 +1,53 @@
...
+		- "easeInOutCirc"
+		- "easeInOutBack"
+		- "easeInOutExpo"
+	<wc-image> - options:
+	- img-alt = "string"
+	- imgs = 
+		'[{"src" : "string", "bp" : number},
+		{"src" : "string", "bp" : number},
+		{ .... },
+		{"src" : "string"}]'
```
Comment (no-creative-name) [inline src/components/WcSlider/scaffolding/WcSlider.hbs]: Pls put the doc for image into the wc image hbs :)

File: src/components/WcSlider/defines.ts [hunk 3]
```diff
@@ -8,7 +8,15 @@
 interface WcSliderProps {
     hasPreview: boolean,
     hasDots: boolean,
-    hasCounter: boolean,
+	hasCounter: boolean,
+	hasAutoplay: autoplayOptions [],
+	slideSpeed: number,
+	easingEffect: string
+}
+interface autoplayOptions{
```
Comment (no-creative-name) [inline src/components/WcSlider/defines.ts]: Interfaces always use PascalCase => "AutoplayOptions"

File: src/components/WcSlider/scaffolding/WcSlider.hbs (hunk 1 above)
Comment (no-creative-name) [inline src/components/WcSlider/scaffolding/WcSlider.hbs]: Also it makes more sense not to put those object into an array. This is better: 
`autoplay-options='{"autoplay": true, "stopOnMouseOver": true, "autoplaySpeed: 7000}'`

File: src/components/WcSlider/WcSlider.ts [hunk 4]
```diff
@@ -183,21 +255,40 @@ class WcSlider extends Component<WcSliderProps, WcSliderState> {
...
+		if (e.type === 'on.lory.resize') {
 			this.dotContainerChildren.forEach((child: HTMLElement) => {
-				child.classList.remove("active");
+				child.classList.remove('active');
 			});
-			this.dotContainer.firstElementChild.classList.add("active");
+			this.dotContainer.firstElementChild.classList.add('active');
 		}
 	}
+	public animationInit() {
```
Comment (no-creative-name) [inline src/components/WcSlider/WcSlider.ts]: initAnimation 🥳

Comment (no-creative-name) [reply src/components/WcSlider/WcSlider.ts]: And btw isn't this rather "initAutoplayAnimation"?

File: src/components/WcSlider/WcSlider.ts [hunk 5]
```diff
@@ -102,31 +148,57 @@ class WcSlider extends Component<WcSliderProps, WcSliderState> {
...
+		this.dotListItem = document.createElement('li');
 		this.counterMax = this.shadowRoot.querySelector(
-			".slider__slideMaxValue"
+			'.slider__slideMaxValue'
 		);
 		this.counterCurr = this.shadowRoot.querySelector(
-			".slider__slideCurrentValue"
+			'.slider__slideCurrentValue'
 		);
+		this.easingEffectsData = JSON.stringify([
```
Comment (no-creative-name) [inline src/components/WcSlider/WcSlider.ts]: Remove JSON.stringify

File: src/components/WcSlider/WcSlider.ts [hunk 6]
```diff
@@ -102,31 +148,57 @@ class WcSlider extends Component<WcSliderProps, WcSliderState> {
...
+				name: 'easeInOutExpo',
+				style: 'cubic-bezier(1, 0, 0, 1)'
+			}
+		]);
 	}
-
-	public handleSlide(e: CustomEvent, slide) {
-		if (e.type === "before.lory.slide") {
+	public setEasingEffect(inputEffect: string) {
+		this.allEasingEffects = JSON.parse(this.easingEffectsData);
```
Comment (no-creative-name) [inline src/components/WcSlider/WcSlider.ts]: Remove JSON.parse

File: src/components/WcSlider/WcSlider.ts [hunk 7]
```diff
@@ -102,31 +148,57 @@ class WcSlider extends Component<WcSliderProps, WcSliderState> {
...
+		this.allEasingEffects.forEach((item: any) => {
+			if (item.name === inputEffect) {
+				inputEffect = item.style;
+			}
+		});
```
Comment (no-creative-name) [inline src/components/WcSlider/WcSlider.ts]: This is easier with Array.find() 🤓
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: src/components/WcSlider/WcSlider.ts [hunk 8]
```diff
@@ -270,8 +274,8 @@ class WcSlider extends Component<WcSliderProps, WcSliderState> {
 			this.dotContainer.firstElementChild.classList.add('active');
 		}
 	}
-	public animationInit() {
-		this.fpsInterval = this.props.hasAutoplay[2].autoplaySpeed;
+	public InitAutoplayAnimation() {
```
Comment (no-creative-name) [inline src/components/WcSlider/WcSlider.ts]: Function names don't follow PascalCase, but camelCase `initAutoplayAnimation`

File: src/components/WcSlider/WcSlider.ts [hunk 9]
```diff
@@ -46,8 +46,9 @@ class WcSlider extends Component<WcSliderProps, WcSliderState> {
 	public then: number = null;
 	public elapsed: number = null;
 	//easing
-	public easingEffectsData: string = null;
+	public easingEffectsData: Array<Object> = null;
 	public allEasingEffects: Array<Object> = null;
```
Comment (no-creative-name) [inline src/components/WcSlider/WcSlider.ts]: "Object" as a type should be avoided, just as "any". Ideally you would add an interface "EasingEffects". Please come to me for further details. :)

**Comment by christophsaile** [comment]
close #39 #40

**Pull request merged**